src/types/ota_config.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef SRC_TYPES_OTA_CONFIG_TYPES_HPP_
+#define SRC_TYPES_OTA_CONFIG_TYPES_HPP_
+
+#include <array>
+#include <cstdint>
+#include <vector>
+
+#include "com_types.hpp"
+
+namespace types {
+
+using ota_config_data = std::vector<std::uint8_t>;
+
+/**
+ * @brief Data structure for address configuration data is as follows:
+ *  ________________________________________
+ * | ID     | Address | PutDataPacketTarget |
+ * |________|_________|_____________________|
+ * | 1 byte | 5 bytes | 1 byte              |
+ * |________|_________|_____________________|
+ */
+
+/// Position for config data id.
+static constexpr std::uint8_t START_OTA_ID = 0;
+/// Begin of address data.
+static constexpr std::uint8_t START_OTA_ADDR_DATA = 1;
+/// End of address data.
+static constexpr std::uint8_t END_ADDR_CONFIG_DATA = 6;
+/// Begin of address target information.
+static constexpr std::uint8_t START_ADDR_TARGET_CONFIG_DATA = 7;
+
+/// ID for address configuration data.
+enum class OtaConfigTypeId : std::uint8_t {
+  CONFIGURE_ADDRESS
+};
+
+/**
+ * @brief Data packet for over the air configuration.
+ *
+ */
+struct OtaConfigPacket : public ComDataPacket {
+  auto GetConfigTypeId() noexcept -> OtaConfigTypeId {
+    return static_cast<OtaConfigTypeId>(data.at(START_OTA_ID));
+  }
+
+  auto EncodeAddressConfigPacket(PutDataPacketTarget target, data_pipe_address address) noexcept -> void {
```

Comment:
The method definition is in the header is risky business. If you can guarantee that this is only included once. It will work, but if it gets included in multiple places ,it could result to a "multiple definition" error at the linking stage. It could also be inline.

Reply:
Will be inline. Fixed.